lib/tenios/api/number.rb

```diff
@@ -0,0 +1,35 @@
+module Tenios
+  module API
+    class Number
+      attr_reader :client
+
+      def initialize(client)
+        @client = client
+      end
+
+      NUMBER_TYPES = %w[GEOGRAPHICAL].freeze
+
+      def order(verification_id:, number_type: NUMBER_TYPES.first, **options)
+        payload = build_payload(verification_id: verification_id, number_type: number_type, **options)
+        client.http_client.post('/number/order', payload).body
+      end
+
+      private
+
+      OPTIONS = %i[
+        link_to_number
+        number_type
+        push_secret
+        push_url
+        verification_id
+      ].freeze
+      private_constant :OPTIONS
+
+      def build_payload(options)
+        { access_key: client.access_key }
+          .merge(options.slice(*OPTIONS))
+          .to_json
```

Comment:
```ruby
options
.slice(*OPTIONS)
.merge(access_key: client.access_key)
.to_json
```

is nicer I think

Reply:
I kept the `access_key` at the top, because the order in the tests matter 😅 (I couldn't figure out how to make it not matter...) and seeing the `access_key` first made sense in my head.

Reply:
hmm, probably you will need to parse the result back into ruby hash in order to get rid of order dependence

Reply:
But it is an `expect().to have_received().with(payload)` where payload is a JSON string...
Maybe there is a better way of doing this 🤔

Reply:
we have configured faraday to expect `json` will it not auto serialize?

https://github.com/carwow/tenios-api-ruby/blob/ab59ddc2831cf98408d4428fc9b1517ff3393464/lib/tenios/api/client.rb#L14-L16

Reply:
Not sure why I started using `to_json`, but removing it worked. 👍 😄